ext/MitsubaLoader/CElementMatrix.h

```diff
@@ -0,0 +1,44 @@
+﻿#ifndef __C_ELEMENT_MATRIX_H_INCLUDED__
+#define __C_ELEMENT_MATRIX_H_INCLUDED__
+
+#include "IElement.h"
+#include "irrlicht.h"
+
+namespace irr { namespace ext { namespace MitsubaLoader {
+
+class CElementMatrix : public IElement
+{
+public:
+	enum class Type
+	{
+		ARBITRARY,
+		TRANSLATION,
+		ROTATION,
+		SCALE
+	};
+
+public:
+	CElementMatrix(CElementMatrix::Type _type)
+		:type(_type) {};
+
+
+	virtual bool processAttributes(const char** _args) override;
+	virtual bool onEndTag(asset::IAssetManager& _assetManager, IElement* _parent) override;
+	virtual IElement::Type getType() const override { return IElement::Type::MATRIX;  };
+	virtual std::string getLogName() const override { return "matrix"; };
+
+	const core::matrix4SIMD getMatrix() const { return matrix; }
+
+private:
+	core::matrix4SIMD matrix;
+	const CElementMatrix::Type type;
```

Comment:
why do you need to track the type?

Reply:
to make downcasting easier, I do a lot of downcasting, especially in processChildData functions.

Just like in this example:
https://github.com/Przemog1/IrrlichtBAW/blob/mmitsuba_loader/ext/MitsubaLoader/CElementShapeOBJ.cpp#L59

Reply:
but matrix is always 4x4 or 3x4 (with implicit 0,0,0,1 last row). This is matrix Type not element type!

Reply:
explain?